### PR TITLE
Moved subscription import/export options to (overflow) menu

### LINKED ISSUE #7453: Subscription import/export options hidden behind secret UI action
### Checklist

- [x] I am using the latest version - 0.21.3 
- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one bug. I will open one issue for every bug report I want to file.

### Steps to reproduce the bug

1. Subscribe to at least one channel.
2. Open the Subscriptions tab.

### Actual behavior

The import/export options get collapsed into the Subscriptions header. From this point, if a user is not already aware of the fact that the header needs to be tapped to expand the import/export options again, then it probably seems like Newpipe only offers this functionality once.

### Expected behavior

The import/export options should be placed somewhere visible, or the action to make them visible should be very obvious to Android users. Or, just don't automatically collapse the header. Let the user figure out how to do that manually.

Another suggestion is given in #7047 (option 2b).

### Context

https://github.com/TeamNewPipe/NewPipe/issues/7326#issuecomment-980270339

### PR DESCRIPTION
**Pull request by litetex**
#### What is it?
- [ ] Bugfix (user facing)
- [x] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
* Moved subscription import/export options to (overflow) menu
* Fixed popup-menu icon colors
* Converted placeholders ``png``s to ``svg``s
  * Required for SubscriptionFragment/Popup (otherwise the Popup-menu uses half of the screen)
  * Size reduction
  * Fixed/Improved some images:
    * Bandcamp: Was facing in the wrong direction and used an incorrect logo
    * Media CCC: Update logo
    * YT: Added NewPipe logo so that it's not just a rectangle
* Cleanup

#### Before/After Screenshots/Screen Record
- Before:
grafik grafik grafik

- After:
grafik grafik grafik grafik grafik

#### Fixes the following issue(s)
- Fixes #7453

#### APK testing 
The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by TiA4f8R** [comment]
Looks good, except on light theme: take a look at the following screenshots of the menu in light theme:

Import from light theme

Export to light theme

The issue doesn't happen on other (and dark) themes.

**Comment by opusforlife2** [comment]
@evilgremlin Please test this.

**Comment by EvilGremlin** [comment]
Looks good

**Comment by opusforlife2** [comment]
In your "Import from"/"Export to" screenshot, the arrows aren't visible. I think they should be visible at all times.

**Comment by opusforlife2** [comment]
Ping, in case this slipped your mind.

**Comment by litetex** [comment]
> Ping, in case this slipped your mind.

Re-Discovered this PR 2 days ago and added it to my TODO list, will have a look after higher priority stuff was done.

<sub>Fell free to close this PR in >2 weeks if still nothing has changed</sub>

**Comment by litetex** [comment]
@TiA4f8R @opusforlife2 
The problems should be fixed now 😄

**Comment by Stypox** [comment]
@litetex please update the YouTube icon as we decided. Other than that, code looks good to me from a first look.

**Comment by litetex** [comment]
@Stypox Done :)

File: app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.kt [hunk 1]
```diff
@@ -150,28 +135,74 @@ class SubscriptionFragment : BaseStateFragment<SubscriptionState>() {
...
+    ): MenuItem {
+        val item = menuItemSupplier.get()
+
+        item.setOnMenuItemClickListener { _ ->
+            onClick.run()
+            true
+        }
+
+        return item
     }
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.kt:206]: No need for `Supplier`s and no need for return values ;-)
```suggestion
    private fun addMenuItemToSubmenu(
        subMenu: SubMenu,
        @StringRes title: Int,
        onClick: Runnable
    ) {
        setClickListenerToMenuItem(subMenu.add(title), onClick)
    }

    private fun addMenuItemToSubmenu(
        subMenu: SubMenu,
        title: String,
        onClick: Runnable
    ) {
        setClickListenerToMenuItem(subMenu.add(title), onClick)
    }

    private fun addMenuItem(
        menuItem: MenuItem,
        onClick: Runnable
    ) {
        menuItem.setOnMenuItemClickListener { _ ->
            onClick.run()
            true
        }
    }
```

**Review by Stypox — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.kt (hunk 1 above)
Comment (litetex) [reply app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.kt]: Fixed

File: app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.kt [hunk 2]
```diff
@@ -150,28 +135,71 @@ class SubscriptionFragment : BaseStateFragment<SubscriptionState>() {
...
-            LocalBroadcastManager.getInstance(activity).unregisterReceiver(subscriptionBroadcastReceiver!!)
-        }
+        addMenuItemToSubmenu(importSubMenu, R.string.previous_export) { onImportPreviousSelected() }
+            .setIcon(R.drawable.ic_backup)
+
+        val services = requireContext().resources.getStringArray(R.array.service_list)
+        for (serviceName in services) {
+            try {
+                val service = NewPipe.getService(serviceName)
+
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.kt:153]: You can use `for (service in ServiceList.all()) {` and avoid handling strings and avoid the `try`. The `service_list` seems to be completely outdated, please remove it altogether (it is not used in any other place):

image

<sup>Sorry for not noticing this earlier</sup>

**Review by Stypox — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.kt (hunk 2 above)
Comment (litetex) [reply app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.kt]: doen

**Review by Stypox — APPROVED** [review]
Code looks good to me. I tested on API 31 importing and exporting things and it works as expected.

The only thing I am still a little bit worried about is discoverability. How can a user know they can import/export subscriptions by tapping on `...`? I would replace the "Nothing here but crickets" with a "You can import and export subscriptions from the three-dot menu". But this would be done in a separate PR.
image

**Comment by opusforlife2** [comment]
> I would replace the "Nothing here but crickets" with a "You can import and export subscriptions from the three-dot menu". But this would be done in a separate PR.

Open an issue until then, so no one forgets it?

**Pull request merged**